src/main/java/org/spdx/sbom/gradle/utils/SpdxDocumentBuilder.java

```diff
@@ -309,13 +294,17 @@ private SpdxPackage createProjectPackage(ResolvedComponentResult resolvedCompone
     return builder.build();
   }
 
-  private Optional<SpdxPackage> createMavenModulePackage(
+  private List<SpdxPackage> createMavenModulePackages(
       ResolvedComponentResult resolvedComponentResult)
       throws InvalidSPDXAnalysisException, IOException {
 
     // if the project doesn't resolve to anything, ignore it
-    File dependencyFile = resolvedExternalArtifacts.get(resolvedComponentResult.getId());
-    if (dependencyFile != null) {
+    Collection<File> dependencyFiles =
+        resolvedExternalArtifacts.get(resolvedComponentResult.getId());
+    if (dependencyFiles == null) return List.of();
```

Comment:
I vaguely recall that `Collections.emptyList()` is slightly more performant over `List.of()`

Reply:
I am not sure that performance is so significant here, however `List.of()` is immutable that's why I used it in the first place.

Reply:
actually yeah this isn't true anymore. It looks like they both return singleton, immutable instances now.